[stable/minio] Bring back config map support

nitisht opened the pull request:
<!--
Thank you for contributing to kubernetes/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/kubernetes/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/kubernetes/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the
history. This will make it easier to identify new changes. The PR will be squashed
anyways when it is merged. Thanks.

For fast feedback, please @-mention maintainers that are listed in the Chart.yaml file.

Please make sure you test your changes before you push them. Once pushed, a CircleCI
will run across your changes and do some initial checks and linting. These checks run
very quickly. Please check the results. We would like these checks to pass before we
even continue reviewing your changes.
-->

**What this PR does / why we need it**: Brings back ConfigMap support to Minio chart

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

unguiculus commented on stable/minio/values.yaml:
```diff
@@ -123,14 +123,34 @@ gcsgateway:
+nasgateway:
+  enabled: false
+  # Number of parallel instances
+  replicas: 4
```
Why would it make sense to have that under `nasgateway` and only apply if `nasgateway.enabled` is `true`?

nitisht replied:
I am not sure if I get the problem here. Do you mean I should use `replicas` instead of `nasgateway.replicas` ?

unguiculus commented on stable/minio/values.yaml:
```diff
@@ -123,14 +123,34 @@ gcsgateway:
+  pv: ""
+  # Storage class of PV to bind. Be default it looks for standard storage class. If the PV uses a different storage class, specify
+  # that here.
+  storageClass: standard
```
Why don't you simply use `persistence.storageClass`? You could avoid the conditional logic.

nitisht replied:
done

nitisht committed:
Bring back config map support to Minio chart

Since ConfigMaps are readonly as of Kubernetes v1.9. Minio chart
copies the config file created via ConfigMap to an emptyDir.

This PR also updates shared mode to NAS gateway mode.
